src/select.ts

```diff
@@ -1,36 +1,47 @@
 import { Injectable } from '@angular/core';
-import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs/Observable';
+import { MemoizedSelector, Store } from '@ngrx/store';
 import { memoize } from './memoize';
```

Comment:
If we are using `MemoizedSelector` can we get rid of `memoize` now?

Reply:
I guess that we can't because we are using different API with store.select . If you want to get rid of it, you should consider changing the API to align with store.select operator and then we can reuse some utility functions from ngrx/store